### PR TITLE
UserInviteForm: move API logic away from components

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Relates to #25117

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/org/api.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { getBackendSrv, locationService } from '@grafana/runtime';
+import { FormModel } from './UserInviteForm';
+
+export const userInviteSubmit = async (formData: FormModel) => {
+  await getBackendSrv().post('/api/org/invites', formData);
```
Comment (natellium) [inline public/app/features/org/api.ts:5]: Aren't we losing some functionality here? I see previously we were triggering an alertError with the message.

File: public/app/features/org/api.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  locationService.push('/org/users/');
```
Comment (natellium) [inline public/app/features/org/api.ts:6]: Not sure this belongs here. I understand api.ts should only contain API calls

File: public/sass/_variables.dark.generated.scss [hunk 3]
```diff
@@ -102,7 +102,7 @@ $text-color: rgb(204, 204, 220);
 $text-color-semi-weak: rgba(204, 204, 220, 0.65);
 $text-color-weak: rgba(204, 204, 220, 0.65);
 $text-color-faint: rgba(204, 204, 220, 0.6);
-$text-color-emphasis: #fff;
+$text-color-emphasis: #FFFFFF;
```
Comment (natellium) [inline public/sass/_variables.dark.generated.scss:105]: Is this on purpose?

**Review by natellium — COMMENTED** [review]
Left a couple of comments.

File: public/app/features/org/api.ts (hunk 1 above)
Comment (thisisobate) [reply public/app/features/org/api.ts:5]: No, we are not. The same alert error is already handled under the hood.

File: public/sass/_variables.dark.generated.scss (hunk 3 above)
Comment (thisisobate) [reply public/sass/_variables.dark.generated.scss:105]: oh no...autogenerated when I started Grafana.

Comment (ashharrison90) [reply public/sass/_variables.dark.generated.scss:105]: probably should be committed to the repo then. i guess someone has forgotten? 🤔

File: public/app/features/org/api.ts (hunk 1 above)
Comment (ashharrison90) [reply public/app/features/org/api.ts:5]: i don't think it is:

| before | after |
| --- | --- |
| image | image |

you could argue we don't need the duplicated notifications. but they say different things, are of different alert levels, and are handled at different places. for example if i did have an smtp server configured but it failed for some other reason, there's no guarantee we'd show an error message anymore...

Comment (thisisobate) [reply public/app/features/org/api.ts:5]: hmm...interesting point. Would reverse it then

File: public/sass/_variables.dark.generated.scss (hunk 3 above)
Comment (thisisobate) [reply public/sass/_variables.dark.generated.scss:105]: yeah...I think so. Seems to be our preferred style of writing hex code.

**Review by natellium — APPROVED** [review]
Looks good to me now 👍

**Review by ashharrison90 — APPROVED** [review]
looks great! 🙌

**Pull request merged**